### PR TITLE
First module/second task

### PR DESCRIPTION
**Pull request by PrimoTibalt**

### CODE REVIEW

File: ModuleOneFirstTaskConsole/ModuleOneFirstTaskConsole.csproj [hunk 1]
```diff
@@ -14,4 +14,8 @@
     </PackageReference>
   </ItemGroup>
```
Comment (alexander-trishin) [inline ModuleOneFirstTaskConsole/ModuleOneFirstTaskConsole.csproj:15]: Please don't forget to include `StyleCop.Analyzers` in all projects.

File: ModuleOneSecondTaskLibrary/GreatingBuilder.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    /// Builds greating text.
+    /// </summary>
+    public class GreatingBuilder
+    {
+        /// <summary>
+        /// Returns greating in format:
+        /// «{current_time} Hello, {name}!».
+        /// </summary>
+        /// <returns>Greating text.</returns>
+        public string Build()
```
Comment (alexander-trishin) [inline ModuleOneSecondTaskLibrary/GreatingBuilder.cs]: This class and method uses only static information, so they should also be static.

File: ModuleOneSecondTaskLibrary/GreatingBuilder.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        {
+            var args = Environment.GetCommandLineArgs();
+            var name = args?.Length > 1 ? args[1] : Environment.UserName;
```
Comment (alexander-trishin) [inline ModuleOneSecondTaskLibrary/GreatingBuilder.cs:22]: A great find for the default value (`Environment.UserName`), but why `args[1]` and not `args[0]`?

File: ModuleOneFirstTaskConsole/Program.cs [hunk 4]
```diff
@@ -17,7 +18,9 @@ public class Program
         /// <param name="args">Console attributes.</param>
         public static void Main(string[] args)
         {
-            ConsolePrinter.PrintGreating();
+            var greatingBuilder = new GreatingBuilder();
+            Console.WriteLine(greatingBuilder.Build());
+            Console.ReadKey();
```
Comment (alexander-trishin) [inline ModuleOneFirstTaskConsole/Program.cs:23]: A `ReadKey` without an informational message can confuse users of your console application.

**Review by alexander-trishin — CHANGES_REQUESTED** [review]
Hey Anton 😃 

Not bad for the first task. Please, fix your code according to my comments and we will close this module.

File: ModuleOneSecondTaskLibrary/GreatingBuilder.cs (hunk 3 above)
Comment (PrimoTibalt) [reply ModuleOneSecondTaskLibrary/GreatingBuilder.cs:22]: The zero-argument is the name of the program. So, we should start with the first one.

File: ModuleOneFirstTaskConsole/ModuleOneFirstTaskConsole.csproj (hunk 1 above)
Comment (PrimoTibalt) [reply ModuleOneFirstTaskConsole/ModuleOneFirstTaskConsole.csproj:15]: I hope I will. Every project has a reference, as I see. Thank you!
image

File: ModuleOneFirstTaskConsole/Program.cs (hunk 4 above)
Comment (PrimoTibalt) [reply ModuleOneFirstTaskConsole/Program.cs:23]: See https://github.com/PrimoTibalt/DotNetMentoringProgram/pull/1/commits/f85449dbc66f4ac637b9b82db9b60246fbadfb02

File: ModuleOneSecondTaskLibrary/GreatingBuilder.cs (hunk 2 above)
Comment (PrimoTibalt) [reply ModuleOneSecondTaskLibrary/GreatingBuilder.cs]: See https://github.com/PrimoTibalt/DotNetMentoringProgram/pull/1/commits/f85449dbc66f4ac637b9b82db9b60246fbadfb02

**Review by alexander-trishin — APPROVED** [review]

**Pull request merged**